### PR TITLE
Added a test-case for Oauth2 client secret revoke/regenerate

### PR DESCRIPTION
**Pull request by IndunilRathnayake**

### CODE REVIEW

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/OAuthAdminServiceTestCase.java [hunk 1]
```diff
@@ -81,9 +82,48 @@ public void testRegisterApplication() throws Exception {
...
+        try {
+            getOauthApplicationState();
+        } catch (Exception e) {
+            Assert.assertFalse(false, "Error when retrieving oauth app state");
+        }
+    }
+
+    @Test(groups = "wso2.is", description = "Testing updating Oauth2 application state", dependsOnMethods = "testRegisterApplication")
+    public void testUpdateOauthAppState() throws Exception {
+
```
Comment (thanujalk) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/OAuthAdminServiceTestCase.java]: Change dependsOnMethods to correct one(testGetOauthAppState)

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/OAuthAdminServiceTestCase.java [hunk 2]
```diff
@@ -81,9 +82,48 @@ public void testRegisterApplication() throws Exception {
...
+                updated = true;
+            }
+        } catch (Exception e) {
+            Assert.assertFalse(false, "Error when retrieving oauth app state");
+        }
+        Assert.assertTrue(updated);
+    }
+
+    @Test(groups = "wso2.is", description = "Testing updating Oauth2 application state", dependsOnMethods = "testRegisterApplication")
+    public void testUpdateOauthSecretKey() throws Exception {
```
Comment (thanujalk) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/OAuthAdminServiceTestCase.java]: Change dependsOnMethods to correct one

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/OAuthAdminServiceTestCase.java [hunk 3]
```diff
@@ -81,9 +82,48 @@ public void testRegisterApplication() throws Exception {
...
+    }
+
+    @Test(groups = "wso2.is", description = "Testing updating Oauth2 application state", dependsOnMethods = "testRegisterApplication")
+    public void testUpdateOauthAppState() throws Exception {
+
+        boolean updated = false;
+
+        try {
+            updateConsumerAppState(OAuth2Constant.OAUTH_APPLICATION_STATE_REVOKED);
+            if(getOauthApplicationState().equalsIgnoreCase(OAuth2Constant.OAUTH_APPLICATION_STATE_REVOKED)) {
```
Comment (thanujalk) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/OAuthAdminServiceTestCase.java]: bring OAuth2Constant.OAUTH_APPLICATION_STATE_REVOKED to the front

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/OAuthAdminServiceTestCase.java (hunk 1 above)
Comment (IndunilRathnayake) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/OAuthAdminServiceTestCase.java]: it's not depends on testGetOauthAppState, it depends on testRegisterApplication.

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/OAuthAdminServiceTestCase.java (hunk 2 above)
Comment (IndunilRathnayake) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/OAuthAdminServiceTestCase.java]: it's not depends on testGetOauthAppState, it depends on testRegisterApplication.

**Pull request merged**